src/debug.cc

```diff
@@ -454,20 +962,57 @@ _db_set_syslog(const char *facility)
 #endif /* LOG_LOCAL4 */
 }
 
-#else /* HAVE_SYSLOG */
+/* SyslogChannel */
+
+static int
+SyslogPriority(const DebugMessageHeader &header)
+{
+    return header.forceAlert ? LOG_ALERT :
+           (header.level == 0 ? LOG_WARNING : LOG_NOTICE);
+}
+
+void
+SyslogChannel::write(const DebugMessageHeader &header, const CompiledDebugMessageBody &body)
+{
+    syslog(SyslogPriority(header), "%s", body.c_str());
+    noteWritten(header);
+}
+
+#else
 
 void
-_db_set_syslog(const char *)
+SyslogChannel::write(const DebugMessageHeader &, const CompiledDebugMessageBody &)
+{
+    assert(!"unreachable code because opened, shouldWrite() are always false");
```

Comment:
This trick will be reported as an error by static analysis. Please just make this a ```{}``` method definition if it has to be defined at all.

Reply:
> This trick will be reported as an error by static analysis.

What makes you think that? What error? FWIW, we are already using the `assert(!"explanation")` trick in `src/acl/Options.h`. It is a well-known trick.

> Please just make this a {} method definition

I think we should assert if a supposed-to-be-unreachable code has been reached, especially when the code is somewhat sensitive and detecting a violation is easy.

Reply:
It was that already existing usage that brought it to my attention. Coverity added an issue. I'm not finding it there right now, but IIRC it was complaining about constant boolean test causing dead code existence.

Reply:
Perhaps Coverity has improved to handle these cases correctly, just like it obviously handles dozens of `assert(false)` cases. I can volunteer to address associated Coverity issues if they (re)surface. Meanwhile, we should make our code worse just because a future Coverity version might mishandle correct code. If that mishandling happens, we will deal with it. Please withdraw your change request.